### PR TITLE
Fix #8722 - HVAC-Diagram.exe broken in v9.5 (Modernize CMake Fortran Settings)

### LINKED ISSUE #8496: EP-Compare can't start (Linux)
Issue overview
--------------

Found 2 problems with `EP-Compare` tool bundled in EnergyPlus installation:

1. It is looking for file `GraphHints.csv` at start-up, which isn't packaged for `UNIX` systems (but is with Mac/Win). See:
https://github.com/NREL/EnergyPlus/blob/879a06490110ad93d32b358c08f76658fe2f38d4/cmake/Install.cmake#L543-L554
2. (low severity) `EP-Compare` binary is linked with 32bits version of libstc++, which is **not** installed by default on recent versions of Linux: 
```
ldd /usr/local/EnergyPlus-9-4-0/PostProcess/EP-Compare/EP-Compare
	[...]
	libstdc++.so.6 => not found
        [...]
```
Workaround: `apt install lib32stdc++6`

It may affect other OS, I didn't test.

### Details

 - Platform: Ubuntu 20.04 LTS
 - Version of EnergyPlus: 9.4.0

### LINKED ISSUE #8722: HVAC-Diagram.exe broken in v9.5
Issue overview
--------------
The HVAC-Diagram.exe that comes with V9.5 installation (official release) does not work properly. In nearly all simulations runs with EP-Launch, the HVAC schematic diagram will not be properly generated.
 
This was tested to be the case in two PCs with Windows 10 installations. A further check with command line shows that the HVAC-Diagram.exe for v9.5 is somehow broken---when it runs (with properly eplusout file in place), it does not give any output. 

A temporary workaround is to copy the HVAC-Diagram.exe from v9.4 installation to place the one in v9.5. The version from v9.4 works fine, and it will also help generate the HVAC schematic diagrams properly for v9.5 runs. 

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): At least on Windows 10, both Pro and Home Editions
 - Version of EnergyPlus (if using an intermediate build, include SHA): v9.5
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

### WIP: I got it building and running fine on Windows, but didn't realize Unix broke...**Edit**: this should be fixed now 

The problem is that on Windows it seems that [CMakeAddFortranSubdirectory](https://cmake.org/cmake/help/latest/module/CMakeAddFortranSubdirectory.html) doesn't know about the target `fortran_project_options` if included via toplevel CMakeList.txt `include(cmake/Fortran.cmake)`. On Unix it seems that each individual subdirectory is aware of it, resulting in a target name clash.

```
CMake Error at cmake/Fortran.cmake:1 (add_library):
  add_library cannot create target "fortran_project_options" because another
  target with the same name already exists.  The existing target is an
  interface library created in source directory
  "/home/julien/Software/Others/EnergyPlus/src/ExpandObjects".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  src/ReadVars/CMakeLists.txt:19 (include)
```

----

- Fix #8722 - HVAC-Diagram.exe broken in v9.5
    - a static flag was no longer passed after #8425 
    - see https://github.com/NREL/EnergyPlus/issues/8722#issuecomment-844207403 for more info 
- Modernize CMake Fortran Settings
    - I did something similar to what @lefticus had started to do in https://github.com/NREL/EnergyPlus/pull/8425/commits/b059d346a72e3e7e10a8a67b60c72100c190d8e3 but reverted.
- Fix #8496 - Install GraphHints.csv on linux as well.
- 
### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by jmarrec** [comment]
The GCC on-premise machine are failing due to a way too old cmake. target_link_options was added in cmake 3.13: https://cmake.org/cmake/help/latest/command/target_link_options.html

**Comment by jmarrec** [comment]
Gneeeeeee why you ruining my life windows?! Previous build had worked, last is failing... In between I only merged develop.
 I'll investigate why windows is failing tomorrow by trying locally...

**Comment by jmarrec** [comment]
So this unrelated build has the same windows error about Transition: https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-c29a7faf1a977cf3396a8c11c50e42b34d2c9646-Win64-Windows-10-VisualStudio-16.html (coming from PR https://github.com/NREL/EnergyPlus/pull/8857)

Same problem on https://github.com/NREL/EnergyPlus/pull/8831

File: src/AppGPostProcess/CMakeLists.txt [hunk 1]
```diff
@@ -10,53 +10,21 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/../../Products")
...
-  endif()
-endif()
-
+set(CMAKE_SKIP_RPATH ${FORTRAN_SKIP_RPATH})
 set(SKIP_BUILD_RPATH TRUE)
+
 add_executable(AppGPostProcess ${SRC})
 set_target_properties(AppGPostProcess PROPERTIES FOLDER Auxiliary)
 
+if(NOT UNIX)  # Need to reinclude it on Windows
```
Comment (Myoldmopar) [inline src/AppGPostProcess/CMakeLists.txt:19]: This is gross.  And a good fix.

File: src/Transition/CMakeLists.txt [hunk 2]
```diff
@@ -24,94 +22,27 @@ set(LIB_SRC
...
 endif()
-
+get_target_property(F_INTERFACE_COMPILE_FLAGS fortran_project_options INTERFACE_COMPILE_OPTIONS)
+message(DEBUG "Transition: F_INTERFACE_COMPILE_FLAGS=${F_INTERFACE_COMPILE_FLAGS}")
+target_link_libraries(TransitionLib PUBLIC fortran_project_options) # ffree-line-length-none
+
+set(VERSIONS
+    1_0_0  1_0_1  1_0_2  1_0_3  1_1_0  1_1_1  1_2_0  1_2_1  1_2_2  1_2_3  1_3_0  1_4_0
+    2_0_0  2_1_0  2_2_0  3_0_0  3_1_0  4_0_0  5_0_0  6_0_0  7_0_0  7_1_0  7_2_0
+    8_0_0  8_1_0  8_2_0  8_3_0  8_4_0  8_5_0  8_6_0  8_7_0  8_8_0  8_9_0
```
Comment (Myoldmopar) [inline src/Transition/CMakeLists.txt:42]: I am compelled to want the 3's, 4, 5, 6, and 7's in their own lines....... :eyes: But I won't hold this up for it.

**Review by Myoldmopar — APPROVED** [review]
Who knew that in 2021, @jmarrec would become a master detective of CMake Fortran subproject quirks?  This is great stuff @jmarrec, we should discuss whether this is being tested exhaustively enough to call it complete.  We can definitely add some tests to the EPTravisTester engine for these auxiliary pieces, but we also want to make sure Decent is doing a _decent_ job of it as well.  From a code perspective this looks good.

**Comment by Myoldmopar** [comment]
I confirmed the package build is good on Linux.  EP-Compare now at least opens up without complaining about the GraphHints being missing.  HVAC Diagram is working fine on Linux, as it was, and for Windows I am relying on the fact that you could reproduce the problem and see the fix.  I ran unit tests on the local build with develop pulled in, no problem.  CI is super happy with this branch.  This is good to go in.  Thanks @jmarrec!  (As a follow-up, we could add some testing of hvac diagram into the EPTravisTester, we'll discuss later)

**Pull request merged**